### PR TITLE
fix: theme flickering

### PR DESCRIPTION
**Pull request by hbjORbj**
## What does this PR do?

- When app theme is not matching user's computer system theme, there's a flickering due to theme mismatch in a page render

## Mandatory Tasks (DO NOT REMOVE)

- [x] I have self-reviewed the code (A decent size PR without self-review might be rejected).
- [x] N/A - I have added a Docs issue [here](https://github.com/calcom/docs/issues/new) if this PR makes changes that would require a [documentation change](https://docs.cal.com/docs). If N/A, write N/A here and check the checkbox.
- [x] I confirm automated tests are in place that prove my fix is effective or that my feature works.

## How should this be tested?

- Are there environment variables that should be set?
- What are the minimal test data to have?
- What is expected (happy path) to have (input and output)?
- Any other important info that could help to test that PR

## Checklist

- I haven't read the [contributing guide](https://github.com/calcom/cal.com/blob/main/CONTRIBUTING.md)
- My code doesn't follow the style guidelines of this project
- I haven't commented my code, particularly in hard-to-understand areas
- I haven't checked if my changes generate no new warnings

### CODE REVIEW

File: apps/web/pages/_document.tsx [hunk 1]
```diff
@@ -67,7 +67,27 @@ class MyDocument extends Document<Props> {
...
+                      requestAnimationFrame(onReady);
+                    }
+                  };
+
+                  requestAnimationFrame(onReady);
+                } catch (e) {
+                  console.error('Error applying theme:', e);
+                }
+              })();
+            `,
```
Comment (sean-brydon) [inline apps/web/pages/_document.tsx:90]: Would this not be better living in a useEffect in _app for example? 
```
  useEffect(() => {
    // Apply theme
    const applyTheme = () => {
      try {
        const theme = localStorage.getItem('app-theme');
        if (theme) {
          document.body.classList.add(theme);
        }
      } catch (e) {
        console.error('Error applying theme:', e);
      }
    };

    applyTheme();
  }, []);
``` 

Should also solve a simular effect?

Comment (emrysal) [reply apps/web/pages/_document.tsx:90]: In my opinion; in this case it's preferable to have it in _document - as this is a bug that only affects full page reloads and has little to do with the app itself (also I tried your proposed solution and it didn't work for me). But the useEffect would run on every page whereas _document only runs during the very first load.

**Review by emrysal — DISMISSED** [review]
This works - but it also breaks the Booker theme for the logged in user - which is meant to take priority over the dashboard theme.
 
image

**Comment by hbjORbj** [comment]
@emrysal Follow up to our discussion: I improved the vanilla JS logic to take the booking page theme into consideration

**Comment by emrysal** [comment]
@hbjORbj FYI it's all good from the booking's end, just waiting for @hariombalhara to confirm embeds.

**Comment by hbjORbj** [comment]
@hariombalhara Can you give this a review?

File: apps/web/pages/_document.tsx [hunk 2]
```diff
@@ -67,7 +67,39 @@ class MyDocument extends Document<Props> {
...
+                    } else {
+                      requestAnimationFrame(onReady);
+                    }
+                  };
+
+                  requestAnimationFrame(onReady);
+                } catch (e) {
+                  console.error('Error applying theme:', e);
+                }
+              })();
```
Comment (hariombalhara) [inline apps/web/pages/_document.tsx:101]: Could we add some unit tests for this. We could make it a function and use fn.String() here.


This logic works in combination with app-providers-app-dir.tsx. So, if possible we should share constants b/w them e.g. 'booking-theme' prefix.


Also, I am not clear why we need to add this logic here. Doesn't AppProviders stuff work now?

Comment (emrysal) [reply apps/web/pages/_document.tsx:101]: No; at the moment there's a brief moment between pageload and app render where it decides on the wrong theme (only when dark mode is selected)

**Review by emrysal — APPROVED** [review]
LGTM 🌑

**Pull request merged**